core/src/main/java/org/wildfly/extras/creaper/core/online/SslOptions.java

```diff
@@ -0,0 +1,384 @@
+        if (this.keyStoreSource == null && this.keyStoreType != KeyStoreType.PKCS11) {
+            return null;
+        }
+
+        final KeyManagerFactory keyManagerFactory = createKeyManagerFactory();
+        KeyStore keyStore = createKeyStore();
+        char[] password = toCharArray(this.keyStorePassword);
+
+        if (this.keyAlias != null) {
+            // creates keystore with single entry to be accessible under given password
+            password = this.keyPassword != null ? this.keyPassword.toCharArray() : password;
+            keyStore = createRepacementKeyStore(keyStore, password);
+        }
+
+        try {
+            keyManagerFactory.init(keyStore, password);
+        } catch (GeneralSecurityException ex) {
+            throw new IllegalArgumentException("Failed to init KeyManagerFactory for " + this.keyStoreSource, ex);
+        }
+
+        return keyManagerFactory.getKeyManagers();
+    }
+
+    private TrustManager[] getTrustManagers() {
+        // pkcs11 providers are configured externally, no truststore source required here
+        if (this.trustStoreSource == null && this.trustStoreType != KeyStoreType.PKCS11) {
+            return null;
+        }
+
+        final TrustManagerFactory trustManagerFactory = createTrustManagerFactory();
+        final KeyStore trustStore = createTrustStore();
+
+        try {
+            trustManagerFactory.init(trustStore);
+        } catch (KeyStoreException ex) {
+            throw new IllegalArgumentException("Failed to init TrustManagerFactory for " + this.trustStoreSource, ex);
+        }
+        return trustManagerFactory.getTrustManagers();
+    }
+
+    private KeyStore createKeyStore() {
+        final KeyStore keyStore = createStore(this.keyStoreType);
+        InputStream keyStoreStream = null;
+
+        if (this.keyStoreSource != null) {
+            keyStoreStream = createStoreStream(this.keyStoreSource);
+        }
+
+        try {
+            keyStore.load(keyStoreStream, toCharArray(this.keyStorePassword));
+        } catch (IOException ex) {
+            if (ex.getCause() instanceof UnrecoverableKeyException) {
+                throw new IllegalArgumentException("Failed to load keystore. Maybe the password is not correct.", ex);
+            }
+            throw new IllegalStateException("Failed to load keystore. Maybe the keystore type is not correct.", ex);
+        } catch (NoSuchAlgorithmException ex) {
+            throw new IllegalStateException("Failed to load keystore.", ex);
+        } catch (CertificateException ex) {
+            throw new IllegalStateException("Failed to load keystore.", ex);
+        } finally {
+            Closeables.closeQuietly(keyStoreStream);
+        }
+
+        return keyStore;
+    }
+
+    /** Creates keystore with single entry (according to key alias) to be accessible under given password. */
+    private KeyStore createRepacementKeyStore(KeyStore keyStore, char[] password) {
```

Comment:
Could you please explain why this is necessary? If I understand correctly, this will just copy one key from the original keystore to a new keystore. So in essence, the new keystore is a clone of the original one, the only difference being that the new keystore only contains one key from the original. Why is that needed?